src/OrbitGl/TrackManager.cpp

```diff
@@ -168,14 +168,20 @@ void TrackManager::SetFilter(const std::string& filter) {
 }
 
 void TrackManager::UpdateVisibleTrackList() {
+  visible_tracks_.clear();
+
   if (filter_.empty()) {
-    visible_tracks_ = sorted_tracks_;
+    std::copy_if(sorted_tracks_.begin(), sorted_tracks_.end(), std::back_inserter(visible_tracks_),
+                 [](const Track* track) { return track->GetVisible(); });
     return;
   }
 
-  visible_tracks_.clear();
   std::vector<std::string> filters = absl::StrSplit(filter_, ' ', absl::SkipWhitespace());
   for (const auto& track : sorted_tracks_) {
+    if (!track->GetVisible()) {
+      break;
```

Comment:
Shouldn't this be `continue` or am I missing something? Are sorted_tracks_ ordered by visibility?

Reply:
(If I'm right, maybe consider adding a test which covers that case.)

Reply:
Yes, good catch. I've extended the tests to cover this and fixed it (see the last commit), PTAL